File: 251004/Kalitsko/3lab/publisher/src/main/java/com/example/rest/mapper/PostMapper.java
package com.example.rest.mapper;

import com.example.rest.dto.PostRequestTo;
import com.example.rest.dto.PostResponseTo;
import com.example.rest.entity.Post;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface PostMapper {
    @Mapping(target = "topic.id", source = "topicId")
    Post toEntity(PostRequestTo request);
    @Mapping(target = "topicId", source = "topic.id")
    PostResponseTo toResponse(Post post);
}

File: 251003/Pankratiev/rest/publisher/src/main/kotlin/bsuir/dc/publisher/mapper/PostMapper.kt
package bsuir.dc.publisher.mapper

import bsuir.dc.publisher.dto.from.PostRequestTo
import bsuir.dc.publisher.dto.to.PostResponseTo
import bsuir.dc.publisher.entity.Issue
import bsuir.dc.publisher.entity.Post

fun PostRequestTo.toEntity(issue: Issue): Post {
    return Post(
        id = this.id,
        issue = issue,
        content = this.content,
    )
}

fun Post.toResponse(): PostResponseTo {
    return PostResponseTo(
        id = this.id,
        issueId = this.issue.id,
        content = this.content
    )
}
